### PR TITLE
Update README with an application/graphql example

### PR DESCRIPTION
**Pull request by papandreou**
As suggested by @helfer in https://github.com/apollographql/graphql-server/issues/217#issuecomment-285598429

TODO:

- [ ] Update CHANGELOG.md with your change (include reference to issue & this PR)
- [x] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass

### CODE REVIEW

**Comment by papandreou** [comment]
Let me know if you think it would be easier on the eyes if the example just `use`d the middleware directly, thus supporting `application/graphql` for other endpoints than `/graphql`:

```js
express()
    .use(bodyParser.json())
    .use(bodyParser.text({ type: 'application/graphql' }))
    .use((req, res, next) => {
        if (req.is('application/graphql')) {
            req.body = { query: req.body };
        }
        next();
    })
    .use('/graphql', graphqlExpress({ schema: myGraphQLSchema }))
    .listen(3000);
```

File: README.md [hunk 1]
```diff
@@ -213,6 +213,33 @@ Despite express-graphql being a reference implementation, GraphQL Server is actu
 
 That said, GraphQL Server is heavily inspired by express-graphql (it's the reference implementation after all). Rather than seeing the two as competing alternatives, we think that they both have separate roles in the GraphQL ecosystem: express-graphql is a reference implementation, and GraphQL Server is a GraphQL server to be used in production and evolve quickly with the needs of the community. Over time, express-graphql can adopt those features of GraphQL Server that have proven their worth and become established more widely.
 
+### application/graphql requests
+
+express-graphql supports the `application/graphql` Content-Type for requests, which is an alternative to `application/json` request with only the query part sent as text. Because GraphQL server does not parse request bodies, you need to handle that in a separate middleware. Here's an example for express:
```
Comment (willxy) [inline README.md]: Might be helpful to say something about, "in the same way that we use bodyparser.json to parse application/json requests for graphql-server, we can use bodyParser.text plus one extra step in order to also parse application/graphql requests." Ie, to clarify that using a bodyparser middleware is not a new thing unique to application/graphql -- in normal graphql-server usage, we're already using a bodyparser middleware to parse application/json.

Comment (papandreou) [inline README.md]: @willxy, thanks for the feedback. I've rephrased the text now :)

**Comment by helfer** [comment]
Thanks @papandreou! 🙂

**Comment by glasser** [comment]
Does this actually let you get the variables and operationName from the URL like express-graphql does?

**Comment by papandreou** [comment]
@glasser, no, the example in its current state does not. TBH I didn't know that express-graphql supported that combo. Should be fairly easy to add it by parsing the query string and adding the `variables` property to `req.body`.

**Pull request merged**